### PR TITLE
sg: Add first integration tests for `sg start`

### PR DESCRIPTION
**Pull request by mrnugget**
This ports the `ttyBuf` I once added to `src-cli` to `lib/tty.Buffer`
and uses it to write some high-level tests for `sg start`.

The tests are pretty easy to write and to provide some confidence that
code works as intended.

We can't use our real configuration if we don't have all the tools
installed, but this test harness should be enough to test some
sg-specific logic.

For example: we should probably write similar tests in the `run` package
and make sure that running multiple commands also works and that
restarting them works, etc.

## Test plan

- Ran `go test` in both of the new packages to test all the tests I added are green

### CODE REVIEW

File: lib/tty/buffer.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+// Buffer is used to test code that uses the `output` library to produce
+// output. It implements io.Writer and can be passed to output.NewOutput
+// instead of stdout/stderr.
+//
+// Buffer parses *most* of the escape codes used by `output` and keeps the
+// produced output accessible through its `Lines()` method.
+//
+// NOTE: Buffer is *not* complete and probably can't parse everything that
+// output produces. It should be extended as needed.
+type Buffer struct {
```
Comment (efritz) [inline lib/tty/buffer.go:16]: Should this type or package be named something related to testing to more obviously convey its usage (outside of this docstring)? `lib/tty` feels like a *very* top-level package.

File: lib/tty/buffer.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	lines [][]byte
+
+	line   int
+	column int
+}
+
+func (t *Buffer) Write(b []byte) (int, error) {
+	var cur int
```
Comment (efritz) [inline lib/tty/buffer.go:24]:
```suggestion
cur := 0
```

**Review by efritz — APPROVED** [review]
Freakin cool.

File: lib/tty/buffer.go (hunk 1 above)
Comment (mrnugget) [reply lib/tty/buffer.go]: Good call. I moved it to `output/outputtest`.

**Pull request merged**